doc/HLD_Version_Numbers.rst

```diff
@@ -0,0 +1,328 @@
+|Stimulus                     |a client sends an rpc to a failed server during recovery    |
++-----------------------------+------------------------------------------------------------+
+|Stimulus source              |server restart                                              |
++-----------------------------+------------------------------------------------------------+
+|Environment                  |recovery                                                    |
++-----------------------------+------------------------------------------------------------+
+|Artifact                     |the server receives the rpc, tagged with REPLAY flag        |
++-----------------------------+------------------------------------------------------------+
+|Response                     |the server compares before-version-counters in the rpc      |
+|                             |with version counters in the units:                         |
+|                             |                                                            |
+|                             |- if RPC.vc < U.vc, the rpc is a resend and should be       |
+|                             |  ignored;                                                  |
+|                             |                                                            |
+|                             |- if RPC.vc == U.vc, the rpc can be applied right away;     |
+|                             |                                                            |
+|                             |- if RPC.vc > U.vc, the rpc should be queued until missing  |
+|                             |  rpc-s are received from other clients, or until recovery  | 
+|                             |  window is closed. As an optimization, if rpc is an        |
+|                             |  "overwrite" update, rather than "diff" update, it can be  |
+|                             |  applied right away                                        |
++-----------------------------+------------------------------------------------------------+
+|Response Measure             |                                                            |
++-----------------------------+------------------------------------------------------------+
+|Questions and Issues         |                                                            |
++-----------------------------+------------------------------------------------------------+
+
+
+***************
+Failures
+***************
+
+This specification is all about handling failures. 
+
+Analysis
+===============
+
+The hybrid operation mode introduced above, disentangles issues of resource management and concurrency control from failure recovery, which traditional session-slot based protocols fail to do. It's also instructive to compare version maintenance with the epoch algorithm: version numbers are, roughly speaking, per-object epochs.
+
+
```

Comment:
The references should be listed here.
But they are missing.

Reply:
Incorporated.